### PR TITLE
Private/mmeeks/zstd rle

### PR DESCRIPTION
**Pull request by mmeeks**
* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: browser/src/layer/tile/Premult.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+	public unpremultiply(data: Uint8Array, byteLength: number, byteOffset: number = 0) : void {
+		if (byteLength > this._bytes.length)
+		{
+			window.console.warn('Unusual - extend array to unpremultiply ' + byteLength + ' vs. ' + this._bytes.length);
+			this.sizeArrays((byteLength + 3) /4);
+		}
+		for (var o8 = 0; o8 < byteLength; o8 += 4) {
+			// premultiplied rgba -> unpremultiplied rgba
```
Comment (caolanm) [inline browser/src/layer/tile/Premult.ts:41]: in core we have a unpremultiply_table table that we initialize once, don't know if that would add any value here too

**Review by caolanm — APPROVED** [review]
looks very shiny

**Comment by caolanm** [comment]
ci didn't seem to trigger properly though, lets try the close and reopen hack

File: browser/src/layer/tile/Premult.ts (hunk 1 above)
Comment (mmeeks) [reply browser/src/layer/tile/Premult.ts:41]: Ooh - good idea :-) at least we have an object to put it on now.

**Comment by caolanm** [comment]
FWIW java/typescript style enforcer is unhappy with format of src/layer/tile/Premult.ts

[warn] src/layer/tile/Premult.ts
[warn] Code style issues found in the above file. Run Prettier with --write to fix.

**Comment by mmeeks** [comment]
Ran prettier on the typescript.

**Comment by mmeeks** [comment]
Now with the endian-swapping headers we need for htons64 on iOS =)

**Comment by timar** [comment]
on iOS:
```
/Users/ci/jenkins/workspace/github_ios_app_co-24.04/kit/Delta.hpp:167:28: error: use of undeclared identifier 'htole64'
                rleLE[i] = htole64(_rleMask[i]);
                           ^
1 error generated.
```

**Pull request merged**